### PR TITLE
Fix product category thumbnail issue

### PR DESCRIPTION
**Pull request by matheuscl**
Fix the issue in #10637.

When WordPress have a trigger to `add-tag` this code can be changed, but for now i think this is the best solution.

### CODE REVIEW

File: includes/admin/class-wc-admin-taxonomies.php [hunk 1]
```diff
@@ -153,6 +153,24 @@ public function add_category_fields() {
 					jQuery( '.remove_image_button' ).hide();
 					return false;
 				});
+				
+				jQuery( document ).ajaxComplete( function( event, request, options ) {
+					if ( request && request.readyState === 4 && request.status === 200
+						&& options.data && options.data.indexOf('action=add-tag') >= 0 ) {
```
Comment (justinshreve) [inline includes/admin/class-wc-admin-taxonomies.php]: There should be spaces between the single quotes here and the parenthesis here.

File: includes/admin/class-wc-admin-taxonomies.php [hunk 2]
```diff
@@ -153,6 +153,24 @@ public function add_category_fields() {
 					jQuery( '.remove_image_button' ).hide();
 					return false;
 				});
+				
+				jQuery( document ).ajaxComplete( function( event, request, options ) {
+					if ( request && request.readyState === 4 && request.status === 200
```
Comment (justinshreve) [inline includes/admin/class-wc-admin-taxonomies.php]: These conditions/checks should be "yoda" style conditions (https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#yoda-conditions).

File: includes/admin/class-wc-admin-taxonomies.php [hunk 3]
```diff
@@ -153,6 +153,24 @@ public function add_category_fields() {
...
+						&& options.data && options.data.indexOf('action=add-tag') >= 0 ) {
+					
+						var res = wpAjax.parseAjaxResponse( request.responseXML, 'ajax-response' );
+						if ( !res || res.errors ) {
```
Comment (justinshreve) [inline includes/admin/class-wc-admin-taxonomies.php]: There should be a space between the `!` and `res`.

File: includes/admin/class-wc-admin-taxonomies.php [hunk 4]
```diff
@@ -153,6 +153,24 @@ public function add_category_fields() {
...
+						}
+						// Clear Thumbnail fields on submit
+						jQuery( '#product_cat_thumbnail' ).find( 'img' ).attr( 'src', '<?php echo esc_js( wc_placeholder_img_src() ); ?>' );
+						jQuery( '#product_cat_thumbnail_id' ).val( '' );
+						jQuery( '.remove_image_button' ).hide();
+						// Clear Display type field on submit
+						jQuery( '#display_type' ).val( '' );
+						return;
+					}
+				});
```
Comment (justinshreve) [inline includes/admin/class-wc-admin-taxonomies.php]: There should be a space between the brace and parenthesis here too.

**Comment by justinshreve** [comment]
Just left a few notes related to coding standards (https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript). Confirmed the fix works.

**Comment by JosephMendez** [comment]
WHY NO SHIPPING METHOD ON REST API 3?????????????????

**Pull request merged**